File: src/main/java/org/zenbot/szolnok/timetable/batch/step/bus/processor/UrlResouceToDocumentJsoupProcessor.java
package org.zenbot.szolnok.timetable.batch.step.bus.processor;

import lombok.extern.slf4j.Slf4j;
import org.jsoup.nodes.Document;
import org.springframework.batch.item.ItemProcessor;
import org.springframework.stereotype.Component;

@Slf4j
@Component
public class UrlResouceToDocumentJsoupProcessor implements ItemProcessor<String, Document> {

    private final JsoupDocumentService jsoupDocumentService;

    public UrlResouceToDocumentJsoupProcessor(JsoupDocumentService jsoupDocumentService) {
        this.jsoupDocumentService = jsoupDocumentService;
    }

    @Override
    public Document process(String url) {
        log.info("Process url=[{}]", url);
         return jsoupDocumentService.getDocument(url);
    }
}

File: szolnok-timetable-batch-bus/src/main/kotlin/org/zenbot/szolnok/timetable/batch/bus/batch/step/bus/processor/UrlResouceToDocumentJsoupProcessor.kt
package org.zenbot.szolnok.timetable.batch.bus.batch.step.bus.processor

import org.jsoup.nodes.Document
import org.slf4j.LoggerFactory
import org.springframework.batch.item.ItemProcessor
import org.springframework.stereotype.Component

@Component
class UrlResouceToDocumentJsoupProcessor(private val jsoupDocumentService: JsoupDocumentService) : ItemProcessor<String, Document> {

    private val log = LoggerFactory.getLogger(UrlResouceToDocumentJsoupProcessor::class.java)

    override fun process(url: String): Document {
        log.info("Process url=[{}]", url)
        return jsoupDocumentService.getDocument(url)
    }
}
